pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package logger
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+// ECSErrorLogger is used to send the error level log to a separated file and rotate
+// error logs when expired, ECS agent will mount the filepath and read it periodically
+type ECSErrorLogger struct {
+	ErrorLogFilePath    string
+	ErrorLogMaxAge      int
+	ErrorLogFileMaxSize int
+
+	errorChannel chan zapcore.Entry
+	queue        []zapcore.Entry
+	Ticker       *time.Ticker
```

Comment:
This also does not need to be exported.  The ticker should be started in the initializer or in `Run()`.